components/ai_chat/core/browser/ollama/ollama_service.cc

```diff
@@ -0,0 +1,275 @@
+    net::DefineNetworkTrafficAnnotation("brave_leo_assistant_ollama_models", R"(
+        semantics {
+          sender: "Brave Leo Assistant"
+          description:
+            "Fetch available models from local Ollama instance for chat."
+          trigger:
+            "User enables Ollama fetching in Leo Assistant settings."
+          data:
+            "HTTP request to localhost:11434/api/tags for models."
+          destination: LOCAL
+        }
+        policy {
+          cookies_allowed: NO
+          setting: "This feature can be disabled in Leo Assistant settings."
+        })");
+
+constexpr net::NetworkTrafficAnnotationTag kOllamaModelDetailsAnnotation =
+    net::DefineNetworkTrafficAnnotation(
+        "brave_leo_assistant_ollama_model_details",
+        R"(
+        semantics {
+          sender: "Brave Leo Assistant"
+          description:
+            "Fetch detailed information for a specific Ollama model."
+          trigger:
+            "User enables Ollama fetching in Leo Assistant settings."
+          data:
+            "HTTP POST request to localhost:11434/api/show with model name."
+          destination: LOCAL
+        }
+        policy {
+          cookies_allowed: NO
+          setting: "This feature can be disabled in Leo Assistant settings."
+        })");
+
+// Max download sizes for Ollama API responses
+constexpr size_t kConnectionCheckMaxSize = 1024;   // 1KB for connection check
+constexpr size_t kModelListMaxSize = 1024 * 1024;  // 1MB for model list
+constexpr size_t kModelDetailsMaxSize = 1024 * 1024;  // 1MB for model details
+
+}  // namespace
+
+OllamaService::OllamaService(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : url_loader_factory_(std::move(url_loader_factory)) {}
+
+OllamaService::~OllamaService() = default;
+
+void OllamaService::BindReceiver(
+    mojo::PendingReceiver<mojom::OllamaService> receiver) {
+  receivers_.Add(this, std::move(receiver));
+}
+
+void OllamaService::IsConnected(IsConnectedCallback callback) {
+  auto request = std::make_unique<network::ResourceRequest>();
+  request->url = GURL(mojom::kOllamaBaseUrl);
+  request->method = "GET";
+
+  auto loader = network::SimpleURLLoader::Create(std::move(request),
+                                                 kOllamaConnectionAnnotation);
+
+  auto* loader_ptr = loader.get();
+  loader_ptr->DownloadToString(
+      url_loader_factory_.get(),
+      base::BindOnce(&OllamaService::OnConnectionCheckComplete,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback),
+                     std::move(loader)),
+      kConnectionCheckMaxSize);
+}
+
+void OllamaService::OnConnectionCheckComplete(
+    IsConnectedCallback callback,
+    std::unique_ptr<network::SimpleURLLoader> loader,
+    std::optional<std::string> response) {
+  bool connected = response &&
+                   response->find("Ollama is running") != std::string::npos &&
```

Comment:
This seems a bit hacky and maybe the text changes depending on locale? Is a 200 response code not sufficient?

Reply:
https://github.com/brave/brave-core/pull/31568/commits/9b21a9d1df1141048c1d8e957b484889d77eec1e